src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js

```diff
@@ -59,24 +64,36 @@ const Dropdown = ({ dropdownListRef, value, options, onChange, small, single, ex
             </li>
         );
     };
-    const renderItemsList = () => {
+    const calculateAndSetItemsListStyles = () => {
         const itemsStyles = {};
+        const { width, left, top, height, bottom } = containerRef.current.getBoundingClientRect();
+
+        itemsStyles.width = width;
+        itemsStyles.left = left;
+
+        if (window.innerHeight - bottom > MIN_ITEMS_LIST_HEIGHT) {
+            itemsStyles.top = top + height + ITEMS_LIST_WIDGET_MARGIN;
+            itemsStyles.maxHeight = window.innerHeight - bottom - ITEMS_LIST_SITE_MARGIN;
+        } else {
+            const headerContainer = document.querySelector('.ibexa-main-header');
```

Comment:
Note: we were discussing why ibexa-main-header offsetHeight was introduced so if you also wonder, you can find the explanation in this comment: https://github.com/ibexa/admin-ui/pull/1183#discussion_r1496978197